server/src/main/java/org/elasticsearch/index/mapper/ContentPath.java

```diff
@@ -15,6 +15,7 @@ public final class ContentPath {
     private final StringBuilder sb;
 
     private int index = 0;
+    private int sbIndex = 0;
```

Comment:
I think we can get rid of this additional index if we change the remove method so it resets the strin builder length according to the length of the element removed.